rpyc/rpyc_client/command_handlers.py

```diff
@@ -91,84 +82,108 @@ def _execute(self):
                 process_id = proc.pid
                 process_list += [{'name': process_name, 'pid': process_id}]
             except (psutil.NoSuchProcess, psutil.AccessDenied, psutil.ZombieProcess):
-                pass
+                pass  # we can't access these processes
 
-        return process_list, None
+        return process_list
 
 
 class FileStatHandler(CommandHandler):
-    def _parse_input(self, split_input, additional_input):
-        if len(split_input) != 2:
-            raise TypeError('expecting one argument')
-        return {'path': split_input[1]}
-
-    def _execute(self, path):
+    def execute(self, command_with_arguments):
+        path = parse_single_path_argument(command_with_arguments)
         remote_os = self.rpyc_conn.modules.os
-        return remote_os.stat(path), None
+        return remote_os.stat(path)
 
 
 class KillProcessHandler(CommandHandler):
-    def _parse_input(self, split_input, additional_input):
-        if len(split_input) < 2:
+    def __init__(self, rpyc_conn, created_processes):
+        super().__init__(rpyc_conn)
+        self.created_processes = created_processes
+
+    def __parse_input(self, command_with_arguments):
+        parser = argparse.ArgumentParser(command_with_arguments[0])
+        parser.add_argument('path')
+        args = parser.parse_args(command_with_arguments[1:])
+        return args.path
+
+    def execute(self, command_with_arguments):
+        if len(command_with_arguments) < 2:
             raise TypeError('expects at least one argument')
 
-        if len(split_input) == 3:
-            if not split_input[1].startswith('-'):
-                raise ValueError("kill signal number should start with '-'")
-            return {'signal': int(split_input[1][1:]), 'pid': int(split_input[2])}
-        elif len(split_input) == 2:
-            return {'pid': int(split_input[1])}
+        if len(command_with_arguments) == 3:
```

Comment:
these should have been done with argparse, since you are already using it
argparse can tell you if '-r' was passed with store_true
you can also add a '--all' to kill all

you shouldn't count the number of argmuents yourself after using argparse

Reply:
tried to make it more clear